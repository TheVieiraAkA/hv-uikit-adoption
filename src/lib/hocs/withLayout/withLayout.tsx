/**
 * Copyright (c) 2019 Hitachi Vantara Corporation.
 *
 *  The copyright to the computer software herein is the property of
 *  Hitachi Vantara Corporation. The software may be used and/or copied only
 *  with the written permission of Hitachi Vantara Corporation or in accordance
 *  with the terms and conditions stipulated in the agreement/contract
 *  under which the software has been supplied.
 */

import React from "react";
import { Header, Footer } from "components/layout";

const getStyles = (hasFooter: boolean) => ({
  content: {
    marginTop: 50, // header height
    padding: `${hasFooter ? 0 : "30px 15px"}` // grid padding
  }
});

const withLayout = <P extends {}>(
  Component: React.ComponentType<P>,
  hasFooter = false
): React.FC<P> => props => {
  const { content } = getStyles(hasFooter);

  return (
    <>
      <Header />
      <section id="content" style={{ ...content }}>
        <Component {...(props as P)} />
      </section>
      {hasFooter && <Footer />}
    </>
  );
};

export default withLayout;
