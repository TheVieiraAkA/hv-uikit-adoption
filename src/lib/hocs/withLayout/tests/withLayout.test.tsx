/**
 * Copyright (c) 2019 Hitachi Vantara Corporation.
 *
 *  The copyright to the computer software herein is the property of
 *  Hitachi Vantara Corporation. The software may be used and/or copied only
 *  with the written permission of Hitachi Vantara Corporation or in accordance
 *  with the terms and conditions stipulated in the agreement/contract
 *  under which the software has been supplied.
 */

import { shallow } from "enzyme";
import React from "react";
import withLayout from "../withLayout";

describe("withLayout", () => {
  let wrapper;

  const checkComponent = (Component: React.ComponentType) => {
    wrapper = shallow(<Component />);
    expect(wrapper).toMatchSnapshot();
  };

  it("is rendered correctly", () => {
    checkComponent(withLayout(React.Component));
    checkComponent(withLayout(React.Component, true));
  });
});
