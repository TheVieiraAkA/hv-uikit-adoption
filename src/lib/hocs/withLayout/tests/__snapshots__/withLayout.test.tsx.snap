// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withLayout is rendered correctly 1`] = `
<div
  style={
    Object {
      "paddingTop": "50px",
      "zIndex": -1,
    }
  }
>
  <WithStyles(Connect(Header)) />
  <div
    style={
      Object {
        "minHeight": "calc(100vh - 50px)",
        "padding": "30px 15px",
      }
    }
  >
    <Component />
  </div>
</div>
`;

exports[`withLayout is rendered correctly 2`] = `
<div
  style={
    Object {
      "paddingTop": "50px",
      "zIndex": -1,
    }
  }
>
  <WithStyles(Connect(Header)) />
  <div
    style={
      Object {
        "minHeight": "calc(100vh - 90px)",
        "padding": "30px 15px",
      }
    }
  >
    <Component />
  </div>
  <WithStyles(Footer) />
</div>
`;
